src/docs/arch-rts.dox

```diff
@@ -1,113 +1,114 @@
+3. The associated files are missing.
+4. The associated files are corrupted.
+5. It is dedicated to @ref arch-metadata "metadata"
+6. It is a logged table. RTS is not necessary for those as their updates are stable when the
+transaction gets committed.
+7. It has no aggregated time window. It is worth noting that the tables with no aggregate time
+window are only created in versions older than MongoDB 4.4
 
-To perform rollback to stable, there shouldn't be any transaction activity happening
-in the WiredTiger.
+For each selected table, its pages are read into the cache. The unstable in-memory updates and the
+unstable historical versions from the history store are removed. The on-disk version is replaced by
+the latest stable in-memory update if any, otherwise by the latest stable version from the history
+store. If there is no stable in-memory updates and no stable version in the history store, the data
+store entry is completely removed. In the situation where the table has no timestamped updates, all
+the historical versions related to this table are removed from the history store as they are no
+longer required and the on-disk value is preserved.
 
-@section rts-table-check Checks performed on a table by rollback to stable
+Finally, RTS is triggered on startup, shutdown or when the application is initiated. It is worth
+noting that it cannot run if there is any transaction activity happening in WiredTiger.
 
-Rollback to stable consider a table to be processed for rollback based on the following
-conditions.
-1. Table is modified
-2. The checkpoint durable start/stop timestamp is greater than the rollback timestamp.
-3. There is no durable timestamp in any checkpoint.
-4. Has prepared updates
-5. Has updates from transactions greater than checkpoint snapshot (only in restart phase)
+@section rts-abort-updates How RTS aborts unstable updates
 
-There are some special conditions where the table is skipped for rollback to stable.
-1. Empty file
-2. Table has timestamp updates but there is no stable timestamp set.
-3. Files that don't exist
-4. Files that are corrupted.
+For in-memory updates, RTS traverses all the updates defined in the update list and aborts every
+update until a stable is found. To remove any key from the table, RTS adds a globally visible
+tombstone to the key update list and this key gets removed later during reconciliation.
 
-If the table has no timestamp updates, then the history store table is
-scanned to remove any historical versions related to this table as these
-older versions no longer required.
+For on-disk values, if the start time pair is not stable and the update list does not contain any
+stable update, RTS searches the history store to find a stable update to replace the on-disk version
+with. If the history store does not contain any stable update, the on-disk key is removed. If the
+stop time pair exists and it is not stable, the on-disk update is restored into the update list.
```

Comment:
we now use time point not time pair

Reply:
I renamed all the `time pair`s into `time point`s, is that correct?